rasterio/_crs.pyx

```diff
@@ -92,6 +92,25 @@ class _CRS(UserDict):
             _safe_osr_release(osr_crs1)
             _safe_osr_release(osr_crs2)
 
+    @property
+    def epsg(self):
+        """The epsg code of the CRS
+
+        Returns
+        -------
+        str
+        """
+        cdef OGRSpatialReferenceH osr = NULL
+
+        try:
+            osr = osr_from_crs(self)
+            if OSRAutoIdentifyEPSG(osr) == 0:
+                epsg_code = OSRGetAuthorityCode(osr, NULL)
+                return epsg_code.decode('utf-8')
```

Comment:
@snowman2 would you be willing to have this return an int that we could plug back into `CRS.from_epsg()` ?

Reply:
Sure, not a problem. What about where it returns None?

Reply:
Also, I would prefer a `to_epsg()` method over an `epsg` property, since in the space of all coordinate reference systems, only a few map to the EPSG definitions. Whereas with the help of `+wktext` all PROJ-based CRS have a WKT representation.

Reply:
Should it return -1? Or would you prefer it to raise a custom exception?

Reply:
I just looked at from_epsg and it looks like it can take and int or a str.

Reply:
I think returning None is fine. Thanks!

Reply:
Right, from_epsg does take a str, but one of the first things we do is `int(code)` to validate, so lets return int (or None) from this new method.